062-support-tiered-storage.md

```diff
@@ -0,0 +1,86 @@
+#  Support tiered storage natively in Strimzi
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively via Strimzi.
+
+## Motivation
+
+Supporting tiered storage via Strimzi allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration for `RemoteStorageManager`, users are able to adopt tiered storage feature on Strimzi easily.
+
+For simplicity, the proposal doesn't add support to customize `RemoteLogMetadataManager`. If tiered storage is enabled, the default implementation class `TopicBasedRemoteLogMetadataManager` in Kafka will be used.
+
+This proposal focuses on tiered storage support with a custom plugin brought in by the user. Other proposals in the future might add some built-in support for selected storage implementations such as S3. It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
+
+## Proposal
+
+Add a `spec.kafka.tieredStorage` field in the `Kafka` CR, which would support configuration for all tiered storage related configuration.
+
+The `tieredStorage` config contains configuration for some selected configuration of tiered storage feature. Most importantly, the configuration for `RemoteStorageManager` is needed.
+
+```
+kafka:
+  tieredStorage:
+      type: custom
+      rsm:
+        type: custom
+        class.name:
+        class.path:  
+        impl.prefix: rsm.config.
```

Comment:
Some thoughts:
* Maybe you can flatten the structure
* You also do not need the `type` twice
* The fields that will be part of the API should use `camelCase`
* Not sure if it is clear what is being configured. So I guess the names can be either `class` / `classPath` or `storageManagerClass` / `storageManagerClassPath`
* While I'm not sure if we need a dedicated `config` section for the RLMM configured mostly by Strimzi (which I raised previously), it might make sense to have it for the storage manager. In such case, the prefix might not be needed?

```suggestion
      type: custom
      storageManagerClass: ...
      storageManagerClassPath: ...
      config:
        # ...
```

Reply:
Addressed other comments. For the last one, do you mean having all the config for RSM (these config will include configuration for the plugin library as well) in the config session here? I am fine with it but I am curious the rationale on not having such config for RLMM but keep it for RSM. I prefer to keep them consistent if possible.